### PR TITLE
[timeseries] Remove statsmodels models

### PR DESCRIPTION
**Pull request by shchur**
*Description of changes:*
- Remove all models from `statsmodels` since these are (1) slower than their counterparts from StatsForecast and (2) not used in any presets.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by melopeo** [comment]
Quick question: this PR is mainly addressing to move models `ARIMAModel, ETSModel, Theta` from `statsmodels` to `statsforecast`. This seems to be suggested by [this](https://github.com/autogluon/autogluon/pull/3513/files#diff-c4e23c6054af5bb74200daf97e42d6b78cee2e4b3efce23c32b1f9d9e1667ef0L5) line. Is this correct?

File: timeseries/src/autogluon/timeseries/models/local/statsforecast.py [hunk 1]
```diff
@@ -178,6 +246,41 @@ def _get_model_type(self):
         return AutoETS
 
 
+class ETSModel(AutoETSModel):
```
Comment (melopeo) [inline timeseries/src/autogluon/timeseries/models/local/statsforecast.py:249]: Why is `ETSModel` designed with parent class `AutoETSModel`, whereas `ARIMAModel` has as parent class `AbstractStatsForecastModel`?

File: docs/tutorials/timeseries/index.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ and does not require you to manually deal with cumbersome issues like model sele
 
 Under the hood, AutoGluon combines various state of the art forecasting algorithms.
 These include established statical methods like ETS and ARIMA from
-[`StatsForecast`](https://github.com/Nixtla/statsforecast) and [`statsmodels`](https://statsmodels.org/) libraries,
+[`StatsForecast`](https://github.com/Nixtla/statsforecast),
```
Comment (melopeo) [inline docs/tutorials/timeseries/index.md:9]: My apologies for this question. Is there anywhere in AGTS documented what are the models available to the user?

Comment (melopeo) [reply docs/tutorials/timeseries/index.md:9]: Ah, it is here!
http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-3513/426f107/tutorials/timeseries/forecasting-model-zoo.html

File: timeseries/src/autogluon/timeseries/models/local/statsforecast.py (hunk 1 above)
Comment (shchur) [reply timeseries/src/autogluon/timeseries/models/local/statsforecast.py:249]: StatsForecast doesn't have a dedicated class for the ETS model but has one for ARIMA. The `ETSModel` class is [deprecated](https://github.com/Nixtla/statsforecast/blob/main/statsforecast/models.py#L828) and it's just a wrapper for AutoETS (similar to our code).

My idea was to provide `ETSModel` as a convenience for users who used the `ETSModel` from statsmodels before.

File: docs/tutorials/timeseries/index.md (hunk 2 above)
Comment (shchur) [reply docs/tutorials/timeseries/index.md:9]: Thanks, that's a really good point! I will remove the references from Model Zoo as well.

Comment (shchur) [reply docs/tutorials/timeseries/index.md:9]: Actually, it seems that the references to the models got updated automatically and there is nothing to change there 😄

Comment (melopeo) [reply docs/tutorials/timeseries/index.md:9]: Yes, it is interesting :)

**Pull request merged**